src/components/tl/ucp/reduce_scatter/reduce_scatter_knomial.c

```diff
@@ -15,42 +15,187 @@
         task->reduce_scatter_kn.phase = _phase;                                \
     } while (0)
 
-static inline void get_sbuf_rbuf(ucc_tl_ucp_task_t *task, size_t block_count,
-                                 void **sbuf, void **rbuf)
+#define GET_COUNT(_args)                                                       \
+    ({                                                                         \
+        size_t _count = 0;                                                     \
+        switch ((_args)->coll_type) {                                          \
+        case UCC_COLL_TYPE_ALLREDUCE:                                          \
+            _count = (_args)->dst.info.count;                                  \
+            break;                                                             \
+        case UCC_COLL_TYPE_REDUCE_SCATTER:                                     \
+            _count = UCC_IS_INPLACE(*(_args))                                  \
+                         ? (_args)->dst.info.count                             \
+                         : (_args)->src.info.count;                            \
+            break;                                                             \
+        default:                                                               \
+            break;                                                             \
+        }                                                                      \
+        _count;                                                                \
+    })
+
+#define GET_DT(_args)                                                          \
```

Comment:
Up to you but this macro is only used once so not necessarily needed

Reply:
right, this is just for future implementation of reduce scatterv knomial